File: challenge/second/python/001.py
from bitcoinrpc.authproxy import AuthServiceProxy

rpc_connection = AuthServiceProxy(
    "http://user_013:nml3pjkYBusJ@217.76.54.77:8332")

# Equivalent to: bitcoin-cli getblockhash 654321
block_hash = rpc_connection.getblockhash(654321)
print(block_hash)

File: challenge/second/rust/src/001.rs
use bitcoincore_rpc::RpcApi;
use rust::get_rpc_from_config;

fn main() -> Result<(), Box<dyn std::error::Error>> {
    // Connect to the RPC endpoint with credentials
    let rpc = get_rpc_from_config().expect("Failed to connect to Bitcoin Core RPC");

    // Equivalent to: bitcoin-cli getblockhash 654321
    let block_hash = rpc
        .get_block_hash(654321)
        .expect("Failed to get block hash");
    println!("{}", block_hash);
    Ok(())
}
/*
# What is the hash of block 654,321?
bitcoin-cli getblockhash 654321
# 000000000000000000058452bbe379ad4364fe8fda68c45e299979b492858095
*/
